ap/production/features.py

```diff
@@ -8,6 +8,8 @@
 from ap.util import maybe_import
 from ap.columnar_util import EMPTY, set_ak_column
 
+from ap.selection.test import jet_energy_shifts
```

Comment:
Actually, now I'm wondering if there is a better way to do this that avoids creating a pseudo-selector (particularly now that the  `variables` is a producer and lives in a completely different module). @riga Ideas?

Reply:
I think the pseudo selector is fine, but after thinking about it some more, I think I wouldn't add `jet_energy_shifts` to the "shifts" sets of all jet-related selectors. At the point where selectors are called, the jer and jes columns (nominal + shifts) are already created, so it's just about declaring the shifts once, which probably should happen in the "main" selector. This will cause the full set of shifts being propagated to the SelectEvents task, which sets the aliases accordingly, and all selectors will receive the varied columns. Or maybe I'm missing something? What do you think?

Reply:
I see what you mean. My intention was to declare the shifts as close as possible to the place they are actually needed. The advantage of that is that the individual selectors then are in principle independent and can be reused in case other "main" selectors are implemented without having to specify the shifts again. But maybe this is not taking into account all the use cases.

Since in the mean time we also discussed introducing a distinction between "exportable" selectors that can be used by tasks and "non-exportable" ones that are called as dependencies, maybe this will have to be refactored annyway. So at least for now I've moved the shifts to the main selector like you suggested.